compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -3332,19 +3340,56 @@ fn print_disambiguation_help<'tcx>(
                 .get(0)
                 .and_then(|ty| ty.ref_mutability())
```

Comment:
The amount of references is not kept in sync between `first_arg_type` and `rcvr_ref` here:

```rs
trait A { fn f(self: &mut &Self); }
trait B { fn f(self: &mut &Self); }

struct S;

impl A for S { fn f(self: &mut &Self) {} }
impl B for S { fn f(self: &mut &Self) {} }

fn main() {
    (&mut &S).f(); // We suggest `A::f(&mut (&mut &S))` but it should be `A::f(&mut &S);`
}

```